benchmark/utils/CxxTests/Subscripts.h

```diff
@@ -0,0 +1,31 @@
+#ifndef BENCHMARK_SUBSCRIPTS_H
+#define BENCHMARK_SUBSCRIPTS_H
+
+struct NonTrivial {
```

Comment:
Sorry, I guess I was probably the one that added this struct, but I don't think shoving a bunch of members into a structs is a great way to test performance. Let's model something more real world, like using a 2D std::vector (or a std::vector<std::string>) or writing your own array type.

Reply:
Ok will do

Reply:
The more I think about it, this might be a little bit contrived, but I think the `std::vector<std::vector<int>>` is better to see a big perf improvement here. Let's make the `std::vector<int>` like 10,000 elements. The parent `std::vector` can just be like 10-100 elements. You can replace `for _ in 0...(N * 10_000)` with `100` or something and then loop over both vectors, putting the final integer into a black hole.

100% optional (as always): you could add another benchmark to test `std::vector<std::string>` too, as that might be a more common case.